lib/hmr.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const sysPath = require('path');
+const logger = require('loggy');
+
+const isEnabled = config => config.hot && config.env.indexOf('production') === -1;
+
+const checkAutoReload = (cfg, plugins) => {
+  const hasAutoReload = plugins.compilers.find(compiler => compiler.brunchPluginName === 'auto-reload-brunch');
+  if (!hasAutoReload) {
+    throw new Error("Hot Module Reloading should only be used with 'auto-reload-brunch'");
+  } else if (!hasAutoReload.supportsHMR) {
+    throw new Error("Currently used version of 'auto-reload-brunch' does not support Hot Module Reloading");
+  }
+};
```

Comment:
This check can't be in the plugin initializer because it needs to be called only after all the plugins are loaded... again, it might make some sense to introduce some private API for a check like this, like `Plugin.prototype.checkInit`, or it might be an overkill...